spring-reactive-modules/spring-5-reactive-3/src/main/java/com/baeldung/webclient/ProductsSlowServiceController.java

```diff
@@ -0,0 +1,21 @@
+package com.baeldung.webclient;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class ProductsSlowServiceController {
+
+    @GetMapping("/slow-service-products")
+    private List<Product> getAllProducts() throws Exception {
```

Comment:
Let's have it throw InterruptedException instead since that's the only non-RuntimeException that it can throw.